src/inference/hmc.jl

```diff
@@ -488,7 +488,7 @@ function observe(
     value,
     vi::VarInfo,
 )
-    return observe(nothing, d, value, vi)
```

Comment:
Why did this change?

Reply:
That was fixed in https://github.com/TuringLang/Turing.jl/pull/995, but when I merged the master branch I noticed that apparently it was reverted by the compiler change https://github.com/TuringLang/Turing.jl/pull/965. However, I guess it's better to move that fix to a separate PR.

Reply:
Yes, that would be great and make PR more clear. Thanks!